cmd/lambda/leases/validate.go

```diff
@@ -57,9 +57,8 @@ func validateLeaseRequest(controller CreateController, req *events.APIGatewayPro
 
 	// Validate requested lease budget amount is less than PRINCIPAL_BUDGET_AMOUNT for current principal billing period
 	usageStartTime := getBeginningOfCurrentBillingPeriod(*controller.PrincipalBudgetPeriod)
-	usageEndTime := time.Date(currentTime.Year(), currentTime.Month(), currentTime.Day(), 23, 59, 59, 0, time.UTC)
 
-	usageRecords, err := controller.UsageSvc.GetUsageByDateRange(usageStartTime, usageEndTime)
+	usageRecords, err := controller.UsageSvc.GetUsageByPrincipal(usageStartTime, requestBody.PrincipalID)
```

Comment:
Why are we only providing the usageStartTime? Does this return usage from startTime - now for the given principle? It seems like keeping the endTime configurable would be desirable.

Reply:
Yes, GetUsageByPrincipal retrieves usage from input start date till now. This method offers more performance tuning than GetUsageByDateRange, since it uses both start date (hash key) and principal ID (sort key).